renpy/display/behavior.py

```diff
@@ -1385,14 +1386,16 @@ def visit(self):
         return [ self.caret ]
 
     def render(self, width, height, st, at):
+        self.st = st
         st -= self.st_base
 
         cr = renpy.display.render.render(self.caret, width, height, st, at)
         rv = renpy.display.render.Render(cr.width, height)
 
-        ttl = self.caret_blink - st % self.caret_blink
+        delay = self.caret_blink / 2.0
+        ttl = self.caret_blink - (delay + st) % self.caret_blink
 
-        if ttl > self.caret_blink / 2.0:
+        if st < delay or (ttl > self.caret_blink / 2.0):
```

Comment:
I believe this is functionally equivalent.
```suggestion
        if st < delay < ttl:
```